lib/Doctrine/Common/Proxy/ProxyGenerator.php

```diff
@@ -20,8 +22,19 @@ class ProxyGenerator
     /**
      * Used to match very simple id methods that don't need
      * to be decorated since the identifier is known.
+     *
+     * @see https://regex101.com/r/u72479/2/
      */
-    const PATTERN_MATCH_ID_METHOD = '((public\s+)?(function\s+%s\s*\(\)\s*)\s*(?::\s*\??\s*\\\\?[a-z_\x7f-\xff][\w\x7f-\xff]*(?:\\\\[a-z_\x7f-\xff][\w\x7f-\xff]*)*\s*)?{\s*return\s*\$this->%s;\s*})i';
+    const PATTERN_MATCH_ID_METHOD = <<<'REGEXP'
+~(
+(public\s+)?
+(function\s+%s\s*\(\)\s*)\s*
+(?::\s*\??\s*((?:\\\\?[\w\\x7f-\\xff][\w\\x7f-\\xff]*)+)\s*)?
+{\s*
+return\s*(\\\$this->%2\$s|new\s+\g{4}\(\\\$this->%2\$s\)|\g{4}::\w+\(\\\$this->%2\$s\));
```

Comment:
Ah wait, that `$s` needs to be interpreted actually, doesn't it?

Reply:
yes, tests say so :)